src/boot/efi/measure.c

```diff
@@ -207,11 +271,16 @@ EFI_STATUS tpm_log_event(uint32_t pcrindex, EFI_PHYSICAL_ADDRESS buffer, size_t
         if (tpm2)
                 err = tpm2_measure_to_pcr_and_event_log(tpm2, pcrindex, buffer, buffer_size, description);
         else {
+                EFI_CC_MEASUREMENT_PROTOCOL *cc;
                 EFI_TCG_PROTOCOL *tpm1;
 
+                cc = cc_interface_check();
+
                 tpm1 = tcg1_interface_check();
                 if (tpm1)
                         err = tpm1_measure_to_pcr_and_event_log(tpm1, pcrindex, buffer, buffer_size, description);
+                else if (cc)
+                        err = cc_measure_to_mr_and_event_log(cc, pcrindex, buffer, buffer_size, description);
                 else {
```

Comment:
hmm, i think this should be nested further in, i.e.

```
tpm2 = tcg2_interface_check();
if (tpm2) 
        err = …
else {
        tpm1 = tcg1_interface_check();
        if (tpm1) 
               err = …;
        else {
               cc = cc_interface_check();
               if (cc) 
                       err = …
               else {
                       /* No active TPM nor CC found, so don't return an error */
                       if (ret_measure)
                               *ret_measured = false;
                       return EFI_SUCCESS;
               }
        }
}

Reply:
(i mean, yeah the nesting gets a bit deep, but 3 is still fine i think. moreover in this cycle i actually want to prep a patch that drops the tpm1.2 code entirely. we never had support in userspace, so we should kill it in sd-boot too. once that's dropped the nesting goes back to 2 again)

Reply:
I was wondering about the importance of tpm1.2 too. Thanks for the feedback, I pushed the suggested version